src/components/TrackTile/TrackerDetails/TrackerDetails.tsx

```diff
@@ -109,62 +109,67 @@ export const TrackerDetails: FC<TrackerDetailsProps> = (props) => {
     [defaultUnit, tracker.units],
   );
 
-  const saveNewValue = useCallback(
-    debounce(async (newValue: number) => {
-      try {
-        const values = todaysValues[metricId] ?? [];
-        const currentValue = values.reduce(
-          (total, { value }) => total + value,
-          0,
-        );
-        const updates: TrackerValue[] = [];
-        const deletes: TrackerValue[] = [];
-        let trackerValueIndex = 0;
-        let delta = convertToStoreUnit(newValue, tracker) - currentValue;
-        while (delta !== 0) {
-          const trackerValue = values[trackerValueIndex];
-          const resourceValue = (trackerValue?.value ?? 0) + delta;
+  const saveNewValue = useMemo(
+    () =>
+      debounce(async (newValue: number) => {
```

Comment:
See [here](https://github.com/facebook/react/issues/19240) for the reason I made this change. Manual testing demonstrated that everything was incrementing/decrementing correctly after making this change